.github/workflows/posix.yml

```diff
@@ -243,6 +315,8 @@ jobs:
 
   upload-coverage:
     needs:
+    - expects-macOS
```

Comment:
We need to add uploading steps (like [line 93](https://github.com/openthread/openthread/pull/6195/files#diff-258c8e6cdf61e87e9d94e06c76fb5e23f602423313565dcb8a46afe648587aceR93)) in `expects-macOS` if we are going to upload its coverage data.

Reply:
I failed to enable coverage on macOS, and we can try it later.

Reply:
Then I think we should remove this dependency first, and add it when coverage on macOS is ready.